src/test/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlagsTest.java

```diff
@@ -70,15 +74,73 @@ void shouldSuccessfullyUpdateCaseFlags() {
+        assertNotNull(response.getData().getRepresentativeFlags());
+        assertNotNull(response.getData().getApplicantFlags());
+    }
+
+    @Test
+    void shouldNotUpdateCaseFlags() {
+        //Given
+        final CaseData caseData = caseData();
+        final CicCase cicCase = CicCase.builder()
+            .fullName(TEST_FIRST_NAME)
+            .applicantFullName(TEST_FIRST_NAME)
+            .representativeFullName(TEST_SOLICITOR_NAME)
+            .build();
+        caseData.setCicCase(cicCase);
+        final CaseDetails<CaseData, State> updatedCaseDetails = new CaseDetails<>();
+        final CaseDetails<CaseData, State> beforeDetails = new CaseDetails<>();
+        beforeDetails.setData(caseData);
+        updatedCaseDetails.setData(caseData);
+        updatedCaseDetails.setId(TEST_CASE_ID);
+        updatedCaseDetails.setCreatedDate(LOCAL_DATE_TIME);
+        systemMigrateCaseFlags.setMigrationFlagEnabled(false);
+
+        //When
+        final AboutToStartOrSubmitResponse<CaseData, State> response =
             systemMigrateCaseFlags.aboutToSubmit(updatedCaseDetails, beforeDetails);
 
         //Then
-        assertThat(response.getData()).isNotNull();
-        assertThat(response.getData().getCaseFlags()).isNotNull();
-        assertThat(response.getData().getSubjectFlags()).isNotNull();
-        assertThat(response.getData().getRepresentativeFlags()).isNotNull();
-        assertThat(response.getData().getApplicantFlags()).isNotNull();
+        assertNotNull(response.getData());
+        assertNull(response.getData().getCaseFlags());
+        assertNull(response.getData().getSubjectFlags());
+        assertNull(response.getData().getRepresentativeFlags());
+        assertNull(response.getData().getApplicantFlags());
     }
 
+    @Test
+    void shouldSuccessfullyUpdateCaseFlagsWithNullValues() {
+        //Given
+        final CaseData caseData = caseData();
+        final CicCase cicCase = CicCase.builder()
+            .fullName(null)
+            .applicantFullName(null)
+            .representativeFullName(null)
+            .build();
+        caseData.setCicCase(cicCase);
+        final CaseDetails<CaseData, State> updatedCaseDetails = new CaseDetails<>();
+        final CaseDetails<CaseData, State> beforeDetails = new CaseDetails<>();
+        beforeDetails.setData(caseData);
+        updatedCaseDetails.setData(caseData);
+        updatedCaseDetails.setId(TEST_CASE_ID);
+        updatedCaseDetails.setCreatedDate(LOCAL_DATE_TIME);
+        systemMigrateCaseFlags.setMigrationFlagEnabled(true);
+
+        //When
+        final AboutToStartOrSubmitResponse<CaseData, State> response =
+            systemMigrateCaseFlags.aboutToSubmit(updatedCaseDetails, beforeDetails);
+
+        //Then
+        assertNotNull(response.getData());
+        assertNull(response.getData().getSubjectFlags());
+        assertNull(response.getData().getRepresentativeFlags());
+        assertNull(response.getData().getApplicantFlags());
```

Comment:
There's a couple problems with this test.
- beforeDetails and updatedCaseDetails have the same caseData so I can't tell if the flags have been successfully updated
- Checking if the flags are not null is not very useful. It would be more useful to test that the expected flags have been returned.

Reply:
For the asserts, since the test is shouldshouldSuccessfullyUpdateCaseFlagsWithNullValues(), subject,applicant,representative  flags will be null only caseflags will not . I think in this test there no value to be returned other than null and not null.